### PR TITLE
Add tests for moving to ReadOnly from various time phases

### PR DESCRIPTION
**Pull request by ktbarrett**
Title says it all.

### CODE REVIEW

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 1]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
     await with_timeout(
         Timer(1, "step"), timeout_time=2.5, timeout_unit="step", round_mode="round"
     )
+
+
+# Riviera VHPI ReadOnly in ValueChange moevs to next time step (gh-4119)
```
Comment (marlonjames) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:342]:
```suggestion
# Riviera VHPI ReadOnly in ValueChange moves to next time step (gh-4119)
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 2]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+@cocotb.test
+async def test_readonly_in_timer(dut):
+    cocotb.start_soon(Clock(dut.clk, 10, "ns").start())
+    await Timer(3, "ns")
+    curr_time = get_sim_time("ns")
+    await ReadOnly()
+    assert get_sim_time("ns") == curr_time
+
+
+# Riviera VHPI ReadOnly in ValueChange moevs to next time step (gh-4120)
```
Comment (marlonjames) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:361]:
```suggestion
# Riviera VHPI ReadOnly in ReadWrite moves to next time step (gh-4120)
```

**Review by marlonjames — COMMENTED** [review]
Wouldn't using `step` as the units in `get_sim_time()` be better?

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 3]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
         Timer(1, "step"), timeout_time=2.5, timeout_unit="step", round_mode="round"
     )
+
+
+# Riviera VHPI ReadOnly in ValueChange moevs to next time step (gh-4119)
+@cocotb.test(expect_fail=SIM_NAME.startswith("riviera") and INTF == "vhpi")
+async def test_readonly_in_valuechange(dut):
+    cocotb.start_soon(Clock(dut.clk, 10, "ns").start())
+    await RisingEdge(dut.clk)
+    curr_time = get_sim_time("ns")
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:347]:
```suggestion
    curr_time = get_sim_time()
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 4]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+    await ReadOnly()
+    assert get_sim_time("ns") == curr_time
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:349]:
```suggestion
    assert get_sim_time() == curr_time
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 5]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+
+
+@cocotb.test
+async def test_readonly_in_timer(dut):
+    cocotb.start_soon(Clock(dut.clk, 10, "ns").start())
+    await Timer(3, "ns")
+    curr_time = get_sim_time("ns")
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:356]:
```suggestion
    curr_time = get_sim_time()
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 6]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+    await ReadOnly()
+    assert get_sim_time("ns") == curr_time
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:358]:
```suggestion
    assert get_sim_time() == curr_time
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 7]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+
+
+# Riviera VHPI ReadOnly in ValueChange moevs to next time step (gh-4120)
+@cocotb.test(expect_fail=SIM_NAME.startswith("riviera") and INTF == "vhpi")
+async def test_readonly_in_readwrite(dut):
+    cocotb.start_soon(Clock(dut.clk, 10, "ns").start())
+    await RisingEdge(dut.clk)
+    curr_time = get_sim_time("ns")
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:366]:
```suggestion
    curr_time = get_sim_time()
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 8]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+    await ReadWrite()
+    assert get_sim_time("ns") == curr_time
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:368]:
```suggestion
    assert get_sim_time() == curr_time
```

File: tests/test_cases/test_cocotb/test_timing_triggers.py [hunk 9]
```diff
@@ -331,3 +337,34 @@ async def test_timer_round_mode(_):
...
+    await ReadOnly()
+    assert get_sim_time("ns") == curr_time
```
Comment (ktbarrett) [inline tests/test_cases/test_cocotb/test_timing_triggers.py:370]:
```suggestion
    assert get_sim_time() == curr_time
```

**Review by marlonjames — APPROVED** [review]

**Pull request merged**